Source/swiftlint/Commands/Baseline.swift

```diff
@@ -0,0 +1,92 @@
+import ArgumentParser
+import Foundation
+import SwiftLintFramework
+
+enum Action: String, ExpressibleByArgument {
+    case report, compare
+}
+
+extension SwiftLint {
+    struct Baseline: ParsableCommand {
+        static let configuration = CommandConfiguration(
+            abstract: "Reports the violations in a baseline (the '\(Action.report)' action) or the violations " +
+                      "that are present in another baseline, but not in the original (the '\(Action.compare)' action)."
+        )
+
+        @Argument(help: "The action to perform on the baseline ('\(Action.report)' or '\(Action.compare)').")
+        var action: Action
+        @Option(help: "The path to the baseline file.")
+        var baseline: String
+        @Option(
+            help: """
+                  The path to a second baseline to compare against the baseline. Violations in \
+                  the second baseline that are not present in the original baseline will be reported \
+                  when the '\(Action.compare)' action is selected.
+                  """
+        )
+        var otherBaseline: String?
+        @Option(
+            help: """
+                  The reporter used to report violations. The 'summary' reporter can be useful to \
+                  provide an overview.
+                  """
+        )
+        var reporter: String?
+        @Option(help: "The file where violations should be saved. Prints to stdout by default.")
+        var output: String?
+
+        func run() throws {
+            switch action {
+            case .report:
+                try report()
+            case .compare:
+                try compare()
+            }
+            ExitHelper.successfullyExit()
+        }
+
+        func validate() throws {
```

Comment:
You could make `Action` two sub-commands to explicitly be able to specify which arguments are allowed in either case.

Reply:
nice - I will give that a go.

Reply:
Ah - that does seem much nicer